pkg/siteconfig/clusterinstance.go

```diff
@@ -118,8 +181,51 @@ func (builder *ClusterInstanceBuilder) Create() (*ClusterInstanceBuilder, error)
 	return builder, err
 }
 
+// Update modifies an existing ClusterInstance on the cluster.
+func (builder *CIBuilder) Update(force bool) (*CIBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating clusterinstance %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		glog.V(100).Infof("clusterinstance %s does not exist in namespace %s",
+			builder.Definition.Name, builder.Definition.Namespace)
+
+		return builder, fmt.Errorf("cannot update non-existent clusterinstance")
+	}
+
+	err := builder.apiClient.Update(context.TODO(), builder.Definition)
+
+	if err != nil {
+		if force {
+			glog.V(100).Infof(
+				msg.FailToUpdateNotification("clusterinstance", builder.Definition.Name, builder.Definition.Namespace))
+
+			err := builder.Delete()
```

Comment:
```suggestion
			err := builder.Delete()
			builder.Definition.ResourceVersion = ""
```
There's an edge case that can cause this function to fail if you don't reset the resource version. Sometimes it's fine but I like to reset it anyway to prevent issues later

Reply:
Okay, updated. Thanks! The proposed change will be in place after submitting a new patch for review.